cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerConfigurationHelper.java

```diff
@@ -0,0 +1,101 @@
+import org.apache.kafka.common.config.ConfigResource;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * This class contains methods for getting current configuration from the kafka brokers asynchronously.
+ */
+public class KafkaBrokerConfigurationHelper {
+
+    private final AdminClientProvider adminClientProvider;
+    protected final SecretOperator secretOperations;
+    private static final Logger log = LogManager.getLogger(KafkaBrokerConfigurationHelper.class);
+
+    public KafkaBrokerConfigurationHelper(AdminClientProvider adminClientProvider, SecretOperator secretOperations) {
+        this.adminClientProvider = adminClientProvider;
+        this.secretOperations = secretOperations;
+    }
+
+    public Future<Admin> adminClient(StatefulSet sts, int podId) {
+        Promise<Admin> acPromise = Promise.promise();
+        String cluster = sts.getMetadata().getLabels().get(Labels.STRIMZI_CLUSTER_LABEL);
+        String namespace = sts.getMetadata().getNamespace();
+        Future<Secret> clusterCaKeySecretFuture = secretOperations.getAsync(
+                namespace, KafkaResources.clusterCaCertificateSecretName(cluster));
+        Future<Secret> coKeySecretFuture = secretOperations.getAsync(
+                namespace, ClusterOperator.secretName(cluster));
+        String hostname = KafkaCluster.podDnsName(namespace, cluster, KafkaCluster.kafkaPodName(cluster, podId)) + ":" + KafkaCluster.REPLICATION_PORT;
+
+        return CompositeFuture.join(clusterCaKeySecretFuture, coKeySecretFuture).compose(compositeFuture -> {
+            Secret clusterCaKeySecret = compositeFuture.resultAt(0);
+            if (clusterCaKeySecret == null) {
+                return Future.failedFuture(Util.missingSecretException(namespace, KafkaCluster.clusterCaKeySecretName(cluster)));
+            }
+            Secret coKeySecret = compositeFuture.resultAt(1);
+            if (coKeySecret == null) {
+                return Future.failedFuture(Util.missingSecretException(namespace, ClusterOperator.secretName(cluster)));
+            }
+
+            Admin ac;
+            try {
+                log.info("about to create an admin client");
+                ac = adminClientProvider.createAdminClient(hostname, clusterCaKeySecret, coKeySecret, "cluster-operator");
+                acPromise.complete(ac);
+            } catch (RuntimeException e) {
+                log.warn("Failed to create Admin Client. {}", e);
+                acPromise.complete(null);
+            }
+            return acPromise.future();
+        });
+    }
+
+    public Future<Map<ConfigResource, Config>> getCurrentConfig(int podId, Admin ac) {
```

Comment:
This method is a bit confusing. It returns Future. But it seems to be synchronous. Which makes me wonder about two things:
* Does it really need to return the future?
* The `describeConfigs` seems to be aynchronous, but you don't use it for the method. So it seems to behave in a blocking way. Should this be wrapped in `executeBlocking`? Or should the KafkaFuture be linked to the Promise? and make this actually asynchronous?